sdk/resourcemanager/Azure.ResourceManager/src/ArmOperation.cs

```diff
@@ -5,11 +5,109 @@
+            return new RehydrationOperation<T>(nextLinkOperation, operationState, operation, options);
+        }
+
+        /// <summary>
+        /// Rehydrates an operation from a <see cref="RehydrationToken"/>.
+        /// </summary>
+        /// <param name="client">The Arm client.</param>
+        /// <param name="rehydrationToken">The rehydration token.</param>
+        /// <param name="options">The Arm client options.</param>
+        /// <returns>The long-running operation.</returns>
+        public static async Task<ArmOperation> RehydrateAsync(ArmClient client, RehydrationToken rehydrationToken, ArmClientOptions options = null)
+        {
+            Argument.AssertNotNull(client, nameof(client));
+            Argument.AssertNotNull(rehydrationToken, nameof(rehydrationToken));
+
+            var nextLinkOperation = (NextLinkOperationImplementation)NextLinkOperationImplementation.Create(client.Pipeline, rehydrationToken);
+            var operationState = await nextLinkOperation.UpdateStateAsync(async: true, default).ConfigureAwait(false);
+            return new RehydrationOperation(nextLinkOperation, operationState);
+        }
+
+        /// <summary>
+        /// Rehydrates an operation from a <see cref="RehydrationToken"/>.
+        /// </summary>
+        /// <param name="client">The Arm client.</param>
+        /// <param name="rehydrationToken">The rehydration token.</param>
+        /// <param name="options">The Arm client options.</param>
+        /// <returns>The long-running operation.</returns>
+        public static async Task<ArmOperation<T>> RehydrateAsync<T>(ArmClient client, RehydrationToken rehydrationToken, ArmClientOptions options = null) where T : notnull
+        {
+
+            Argument.AssertNotNull(client, nameof(client));
+            Argument.AssertNotNull(rehydrationToken, nameof(rehydrationToken));
+
+            bool isResource = IsResource<T>();
+            IOperationSource<T> source = new GenericOperationSource<T>(client, isResource);
+            var nextLinkOperation = (NextLinkOperationImplementation)NextLinkOperationImplementation.Create(client.Pipeline, rehydrationToken);
+            var operation = NextLinkOperationImplementation.Create(source, nextLinkOperation);
+            var operationState = await operation.UpdateStateAsync(async: true, default).ConfigureAwait(false);
+            return new RehydrationOperation<T>(nextLinkOperation, operationState, operation, options);
+        }
+
+        private static bool IsResource<T>() where T : notnull
+        {
+            var isResource = typeof(T).GetConstructor(
+                            BindingFlags.NonPublic | BindingFlags.Instance,
+                            null,
+                            CallingConventions.Any,
+                            new Type[] { typeof(ArmClient), typeof(ResourceIdentifier) },
+                            null) is not null;
```

Comment:
does this have higher performance than we just check the base type of T is `ArmResource`?

Reply:
OK - I saw the implementation in the other place that we are calling this ctor using reflection, therefore I understand that we are ensuring that we have this ctor.
Well but I still have the question: should we check the base class here, and just throw exception if the resource does not have the ctor? Because it is not reasonable that a resource in our generated library does not have the ctor

Reply:
To check T is ArmResource, we need to leverage reflection to get the implemented interfaces, which is 
```csharp
var isResource = typeof(ArmResource).IsAssignableFrom(typeof(T));
```
`Activator.Create` is basically the same as GetConstructor here, which is more light-weight than above method.

Reply:
it is hard to imagine that to query if the interface is on the type is slower than we actually called the ctor and create an instance, especially we could have multiple ctors and we are asking one with the specific signature.